internal/regopolicyinterpreter/logging.go

```diff
@@ -0,0 +1,168 @@
+package regopolicyinterpreter
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"sync/atomic"
+
+	hcsshimLog "github.com/Microsoft/hcsshim/internal/log"
+	"github.com/sirupsen/logrus"
+)
+
+type logLevel struct {
+	level LogLevel
+}
+
+func (ll *logLevel) SetLevel(level LogLevel) {
+	atomic.StoreInt32((*int32)(&ll.level), int32(level))
+}
+
+func (ll *logLevel) Level() LogLevel {
+	return LogLevel(atomic.LoadInt32((*int32)(&ll.level)))
+}
+
+type InterpreterLogger interface {
+	LogInfo(ctx context.Context, msg string, args ...interface{})
+	LogResult(ctx context.Context, rule string, resultSet interface{})
+	LogMetadata(ctx context.Context, data map[string]interface{})
+	Entry(ctx context.Context) *logrus.Entry
+	SetLevel(level LogLevel)
+	Level() LogLevel
+	Close(ctx context.Context) error
+}
+
+type fileLogger struct {
+	*logLevel
+	logFile *os.File
+	logger  *logrus.Logger
+}
+
+var _ InterpreterLogger = (*fileLogger)(nil)
+var _ InterpreterLogger = (*logrusLogger)(nil)
+
+func NewFileLogger(path string, level LogLevel) (InterpreterLogger, error) {
+	file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
Does it make sense to set the log file to be `0644` so that it can only be written to by the owning user?